core/platform/lf_rp2040_support.c

```diff
@@ -214,6 +216,170 @@ int _lf_single_threaded_notify_of_event() {
     sem_release(&_lf_sem_irq_event);
     return 0;
 }
+
+#else // LF_SINGLE_THREADED
+
+#warning "Threaded runtime on RP2040 is still experimental"
+
+/**
+ * @brief Get the number of cores on the host machine.
+ */
+int lf_available_cores() {
+    // Right now, runtime creates 1 main thread and 1 worker thread
+    // In the future, this may be changed to 2 (if main thread also functions
+    // as a worker thread)
+    return 1;
+}
+
+static void *(*thread_1) (void *);
+static void* thread_1_args;
+static int num_create_threads_called = 0;
+static semaphore_t thread_1_done;
+static void* thread_1_return;
+
+#define MAGIC_THREAD1_ID 314159
+
+void core1_entry() {
+    thread_1_return = thread_1(thread_1_args);
+    sem_release(&thread_1_done);
+
+    // infinite loop; core1 should never exit
+    while (1){
+        tight_loop_contents();
+    }
+}
+
+int lf_thread_create(lf_thread_t* thread, void *(*lf_thread) (void *), void* arguments) {
+    // make sure this fn is only called once
+    if (num_create_threads_called != 0) {
+        return 1;
+    }
+    thread_1 = lf_thread;
+    thread_1_args = arguments;
+    num_create_threads_called += 1;
+    sem_init(&thread_1_done, 0, 1);
+    multicore_launch_core1(core1_entry);
+    *thread = MAGIC_THREAD1_ID;
+    return 0;
+}
+
+int lf_thread_join(lf_thread_t thread, void** thread_return) {
+    if (thread != MAGIC_THREAD1_ID) {
+        return 1;
+    }
+    sem_acquire_blocking(&thread_1_done);
```

Comment:
Iirc since the main thread in reactor c just blocks until join completes, this would block core0 from executing anything

Reply:
Yup. I should have mentioned this in the PR description, but I plan to start another PR once this one gets merged. Basically:
 - this PR: basic multicore support, only supports one worker thread. As you said, this is kind of pointless, but it's relatively straightforward.
 - next PR: modify the runtime so that the main thread also runs a worker, instead of just waiting for the worker threads. This will allow 2 workers (one main thread, one additional thread) and finally make the multicore support useful. However, this change may potentially affect all platforms, so I want it to be a seperate PR.

Reply:
Neat! Sounds like a plan :)